### PR TITLE
Add mass fraction diffusion coefficients

### PR DESCRIPTION
**Pull request by totto82**
Adds keywords DIFFAGAS and DIFFAWAT which are equivalent with DIFFCGAS and DIFFCWAT assumes a mass and not mol fraction formulation of diffusion.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this opm-models=868 please

**Comment by totto82** [comment]
jenkins build this opm-models=868 please

File: src/opm/input/eclipse/EclipseState/Tables/TableManager.cpp
Comment (bska) [inline src/opm/input/eclipse/EclipseState/Tables/TableManager.cpp:202]: These would be easier to implement using the `requires`/`prohibits` mechanism of the JSON schema.  One example usage is in [`SWOF`](https://github.com/OPM/opm-common/blob/1224f8c974483a7c23bdfc844906461382d7031a/src/opm/input/eclipse/share/keywords/000_Eclipse100/S/SWOF#L10-L11).

File: opm/input/eclipse/EclipseState/Tables/TableManager.hpp
Comment (bska) [inline opm/input/eclipse/EclipseState/Tables/TableManager.hpp:423]: Does this mean we default to enabling this feature?  If so, I think we should discuss this a bit&ndash;at least if there's computational cost associated to having the feature on.

**Review by bska — COMMENTED** [review]
I'd like to see a few unit tests which demonstrate that we properly reject nonsensical combinations at the parser level.

File: opm/input/eclipse/EclipseState/Tables/TableManager.hpp
Comment (totto82) [reply opm/input/eclipse/EclipseState/Tables/TableManager.hpp:423]: You still need to enable diffusion. This just mean that if you dont set the DIFFAWAT of DIFFAGAS keyword we use the mole fraction formulation.

File: src/opm/input/eclipse/EclipseState/Tables/TableManager.cpp
Comment (totto82) [reply src/opm/input/eclipse/EclipseState/Tables/TableManager.cpp:202]: Thanks. Will update.

**Comment by totto82** [comment]
jenkins build this opm-models=868 please

**Comment by bska** [comment]
jenkins build this opm-models=868 please

**Comment by totto82** [comment]
jenkins build this opm-models=868 please

**Comment by totto82** [comment]
jenkins build this opm-models=868 opm-simulators=5143 please

**Comment by totto82** [comment]
jenkins build this opm-models=868 opm-simulators=5143 please

**Review by bska — COMMENTED** [review]
Thanks a lot for the updates.  This looks good in isolation now and I'm prepared to merge this into the master branch.  Before I do, though, I wonder if this can be merged independently of the downstream changes (OPM/opm-models#868 and OPM/opm-simulators#5143).  If so, I'll go ahead and merge it.  Otherwise, I'll hold off until the other PRs have been reviewed and approved.

**Comment by totto82** [comment]
> Otherwise, I'll hold off until the other PRs have been reviewed and approved.

It should be save to merge this first and then the other PRs in models and simulators. But they need to be merged together. They should be fast to review.

**Review by bska — APPROVED** [review]
> It should be save to merge this first and then the other PRs in models and simulators. But they need to be merged together.

Understood.  Thanks a lot for the information.  I'll merge this into master now.

**Comment by gdfldm** [comment]
> There's one thing I forgot to ask: Is `DIFFAWAT` an OPM extension? If so, the keyword's description might be more properly placed in [`900_OPM/D`](https://github.com/OPM/opm-common/tree/48c513e0c9f9f9fa85f22da41fbf77cc3a531c2a/src/opm/input/eclipse/share/keywords/900_OPM/D) instead.

The commercial simulator doesn't seem to have a DIFFAWAT keyword. However, it does have a DIFFAGAS keyword which is used to define activity corrected gas diffusion coefficients. So, it might be worth considering changing the names of these OPM keywords?

**Pull request merged**
